doc/SAI-Proposal-ICMP-ECHO.md

```diff
@@ -0,0 +1,424 @@
+   |                                                               |
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+   |TLV Type       |   TLV Length (0)                  |           |
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+:
+
+- Cookie
+    4B, processing of this filed is imepplemtation specific.
+- Version
+    4B, set to 0
+- GUID MSB
+    4B, The probing session unique identifier
+- GUID LSB
+    4B, The probing session unique identifier
+- Sequence Number
+    8B, processing of this filed is implementation specific.
+- TLV Type
+    1B, SENTINAL: 0xFF
+- TLV Length
+    2B, set to 0
+
+```
+***GUID:*** Global unique identifier for the ICMP echo session to correlate probe and reply packets with a session, especially when the session is offloaded to hardware, a unique identifier is required.
+
+***Session Cookie:*** The offloading engine interprets this value to decide whether the packet should undergo processing by the ICMP echo session or not. Packets not matching the cookie are punted to NOS. This can also be used to differentiate regular icmp response and icmp link monitoring response.
+
+### 2.3.0 ICMP Echo Session State ###
+
+If a targeted device does not respond to an ICMP echo request within the detection time period, the link is marked "DOWN", If an ICMP echo response is subsequently received, the link is marked "UP".
+
+
+![ICMP probe state machine](figures/sai_icmp_echo_state.png "Figure 1: icmp probe state machine ")__Figure 1: ICMP echo state machine.__
+
+
+## 3.0 Specification ##
+
+### sai.h ###
+New type SAI_API_ICMP_ECHO is added into sai_api_t
+
+### saiswitch.h ###
+
+New attribute SAI_SWITCH_ATTR_ICMP_ECHO_SESSION_STATE_CHANGE_NOTIFY is added into sai_switch_attr_t
+
+
+### New Header saiicmpecho.h ###
+
+#### sai_icmp_echo_session_state_t ####
+This enumeration is used to represent the possible states of an ICMP echo session
+
+SAI_ICMP_ECHO_SESSION_DOWN: Signifies that the ICMP echo session has not received an ICMP echo reply packet within the designated receive interval period.
+
+SAI_ICMP_ECHO_SESSION_UP: Denotes that the ICMP echo session consistently receives ICMP echo reply packets within the specified receive interval period.
+
+#### sai_icmp_echo_session_state_notification_t ###
+This defines state change notification fields
+
+icmp_echo_session_id:  Identifies the ICMP echo session related to the state change notification.
+
+session_state: Reflects the operational state of the ICMP echo session, which can be either SAI_ICMP_ECHO_SESSION_UP or SAI_ICMP_ECHO_SESSION_DOWN.
+
+#### sai_icmp_echo_session_attr_t ####
+This defines icmp echo session attributes. These attributes collectively define the configuration and behavior of ICMP echo sessions in the SAI framework.
+
+- SAI_ICMP_ECHO_SESSION_ATTR_VIRTUAL_ROUTER: Specifies whether hardware lookup is valid, allowing for virtual router configuration.
```

Comment:
should be `SAI_ICMP_ECHO_SESSION_ATTR_HW_LOOKUP_VALID`

Reply:
fixed.